### ISSUE TITLE
add User-Agent header

### ISSUE THREAD

**Issue by ctaggart** [Azure/azure-sdk-for-rust#65]
Add a default [User-Agent header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent). This will allow telemetry to be collected and help build a case for making this SDK more official.

```
User-Agent: <product> / <product-version> <comment>
User-Agent: Mozilla/5.0 (<system-information>) <platform> (<platform-details>) <extensions>
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36
```

## .NET
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/TelemetryPolicy.cs
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/avs/Microsoft.Azure.Management.Avs/tests/SessionRecords/AvsTests/AvsCrud.json#L15-L20
``` json
        "User-Agent": [
          "FxVersion/4.6.28928.01",
          "OSName/Windows",
          "OSVersion/Microsoft.Windows.10.0.19041.",
          "Microsoft.Azure.Management.ResourceManager.ResourceManagementClient/1.6.0.0"
        ],
```

## Python
From the az cli:
```
'User-Agent': 'python/3.6.6 (Windows-10-10.0.19041-SP0) msrest/0.6.13 msrest_azure/0.6.3 avsclient/2020-03-20 Azure-SDK-For-Python AZURECLI/2.5.0 (MSI)'
```

**Comment by ctaggart** [comment]
Based on https://azure.github.io/azure-sdk/general_azurecore.html, I think a user agent example would be:
```
azsdk-rust-mgmt_vmware/0.1.0 (x86_64-pc-windows-msvc)
```

The platform section is the target triple listed in https://doc.rust-lang.org/nightly/rustc/platform-support.html which can be set at compilation time, may be:
``` rust
const PLATFORM: &str = env!("CARGO_BUILD_TARGET");
```

An `application_id` can be prepended.

**Comment by cataggar** [comment]
Fixed by #527/
